src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,42 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  unhide() {
+    this.hidden = false;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore.hidden === false && herbivore.health > 0) {
+      herbivore.health -= 50;
+
+      if (herbivore.health <= 0) {
+        const animalIndex = Animal.alive.indexOf(herbivore);
+
+        Animal.alive.splice(animalIndex, 1);
+      }
+    }
```

Comment:
The `bite` method should be optimized for performance. Using `indexOf` and `splice` to remove an animal from the `Animal.alive` array can be inefficient, especially with larger arrays. Consider using `filter` instead to improve performance and avoid nesting loops.